pkg/cephfs/volumemounter.go

```diff
@@ -88,10 +115,9 @@ func newMounter(volOptions *volumeOptions) (volumeMounter, error) {
 	var chosenMounter string
 
 	for _, availMounter := range availableMounters {
-		if chosenMounter == "" {
-			if availMounter == wantMounter {
-				chosenMounter = wantMounter
-			}
+		if availMounter == wantMounter {
```

Comment:
if kernel version < 4.17 and volOptions.Mounter == kernelMounter, the chosen mounter would be set as kernelMounter, since kernelmounter is an available mounter, right? We don't want this.

Reply:
I thought you said that its required? If user doesn't need Quota feature, they could still use kernel mounter?

Reply:
But then Quota is always set on the directories when PVC gets created. So we rather not use kernel mount irrespective of what user specifies if the kernel version is <= 4.17?

Reply:
"Quotas are cooperative and non-adversarial. CephFS quotas rely on the cooperation of the client who is mounting the file system to stop writers when a limit is reached. A modified or adversarial client cannot be prevented from writing as much data as it needs. Quotas should not be relied on to prevent filling the system in environments where the clients are fully untrusted."

"Quotas are implemented in the kernel client 4.17 and higher. Quotas are supported by the userspace client (libcephfs, ceph-fuse). Linux kernel clients >= 4.17 support CephFS quotas but only on mimic+ clusters. Kernel clients (even recent versions) will fail to handle quotas on older clusters, even if they may be able to set the quotas extended attributes."

Source, https://docs.ceph.com/docs/master/cephfs/quota/#limitations

Reply:
> I thought you said that its required? If user doesn't need Quota feature, they could still use kernel mounter?

Sorry, I don't follow. What did I say was required? 

In CSI is there a concept of volumes with infinite capacity? In that case kernel client don't need to be version >= 4.17.

Reply:
> But then Quota is always set on the directories when PVC gets created. So we rather not use kernel mount irrespective of what user specifies if the kernel version is <= 4.17?

If kernel client version less than 4.17, and quota xattr is set on the subvolume. the kernel client wouldn't respect the quota limit set on the subvolume.